ext/fetch/lib.rs

```diff
@@ -395,11 +395,7 @@ pub async fn op_fetch_send(
   let url = res.url().to_string();
   let mut res_headers = Vec::new();
   for (key, val) in res.headers().iter() {
-    let key_bytes: &[u8] = key.as_ref();
-    res_headers.push((
-      ByteString(key_bytes.to_owned()),
-      ByteString(val.as_bytes().to_owned()),
-    ));
+    res_headers.push((key.as_str().into(), val.as_bytes().into()));
```

Comment:
```suggestion
    res_headers.push((key.as_ref().into(), val.as_bytes().into()));
```
IIUC with `.as_str()` there's a round-trip of decoding and then encoding back to Latin-1.

Reply:
It does not, I just double-checked and `HeaderName` is an enum of Custom/Standard, Custom values are implement via an internal `ByteStr` type with a `str` deref of:
```rs
impl ops::Deref for ByteStr {
    type Target = str;

    #[inline]
    fn deref(&self) -> &str {
        let b: &[u8] = self.bytes.as_ref();
        // Safety: the invariant of `bytes` is that it contains valid UTF-8.
        unsafe { str::from_utf8_unchecked(b) }
    }
}
```

And Standard headers have `.as_str() -> &'static str` compile-time strings, so `.as_str()` is essentially zero-cost

Also if you look at the `AsRef<[u8]>` impl it calls `.as_str()`:
```rs
impl AsRef<[u8]> for HeaderName {
    fn as_ref(&self) -> &[u8] {
        self.as_str().as_bytes()
    }
}
```